exonum/src/node/consensus.rs

```diff
@@ -589,6 +590,36 @@ impl NodeHandler {
         }
     }
 
+    /// Handles raw transactions.
+    #[cfg_attr(feature = "flame_profile", flame)]
+    pub fn handle_txs(&mut self, msg: &TransactionsResponse) {
+        if msg.to() != self.state.consensus_public_key() {
+            error!(
+                "Received request that intended for another peer, to={}, from={}",
+                msg.to().to_hex(),
+                msg.from().to_hex()
+            );
+            return;
+        }
+
+        if !self.state.whitelist().allow(msg.from()) {
+            error!(
+                "Received request message from peer = {} which not in whitelist.",
+                msg.from().to_hex()
+            );
+            return;
+        }
+
+        if !msg.verify_signature(msg.from()) {
+            error!("Received request with incorrect signature, msg={:?}", msg);
+            return;
+        }
+
+        for tx in msg.transactions() {
```

Comment:
Do I understand correctly that this code clones all raw messages? Doesn't look very good, tbh.

Reply:
Yes, you are right. How do you think, how should this be done?